### DISCUSSION TITLE
Can maui  support the UWP-style databind using "x: bind" ?

### DISCUSSION THREAD

**Discussion by DJDQfff** [dotnet/maui#12078] [Ideas]
I am used to write uwp app, and the uwp special databind x:bind is very good to use .

For example:
there is a IEnumberable object in the code-behind cs file:

```csharp
ObservableCollection<string> showlist=new ();
```

There is ListView to show the field showlist.

In MAUI ,if i write :

```xml
 <ListView ItemsSource={binding source={x:Reference showlist}}/>
```

app will show error when running .(error message:can't find the object showlist).
then I need to set the ItemsSource by code in cs file.

```csharp
listview.ItemsSource=showlist;
```

but In UWP it can easily work :

```xml
 <ListView ItemsSource={x:bind showlist}/>
```

There are also others advantage of UWP's databind.

If i want to bind the Path, uwp can show the property that can bind.
but MAUI can't .

Maui only find the property when running application.

Although now maui can use *XamlCompileAttribute* to pre-compile xaml,it can't smartly show the property when writind xaml.

**Comment by andrekoehler** [comment]
I believe MAUI also supports compiled bindings, including syntax completion for properties of the view model inside the XAMl editor: https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/data-binding/compiled-bindings?view=net-maui-8.0

I never used UWP, but it seems like in MAUI you can just use {Binding ...} instead of {x:Bind ...} but you have to set the type of the view model with x:DataType in order to get compile-time binding performance.
